### PR TITLE
 Replace tuples with type union in isinstance or issubclass calls

### PR DESCRIPTION
**Pull request by martinlehoux**
- #2923

### CODE REVIEW

File: crates/ruff/src/rules/pyupgrade/rules/use_pep604_annotation.rs [hunk 1]
```diff
@@ -75,6 +75,32 @@ enum TypingMember {
...
 
+pub fn use_pep604_annotation_for_isinstance_or_issubclass(
+    checker: &mut Checker,
+    expr: &Expr,
+    func: &Expr,
+    args: &[Expr],
+) {
+    match &func.node {
+        ExprKind::Name { id, .. } if id == "isinstance" || id == "issubclass" => {
+            if let Some(types) = args.get(1) {
```
Comment (charliermarsh) [inline crates/ruff/src/rules/pyupgrade/rules/use_pep604_annotation.rs:86]: Nit: check `checker.is_builtin(id)` here.

File: crates/ruff/src/rules/pyupgrade/rules/use_pep604_annotation.rs [hunk 2]
```diff
@@ -75,6 +75,32 @@ enum TypingMember {
     Optional,
 }
 
+pub fn use_pep604_annotation_for_isinstance_or_issubclass(
```
Comment (charliermarsh) [inline crates/ruff/src/rules/pyupgrade/rules/use_pep604_annotation.rs:78]: Sorry to go back on my own suggestion, but on second thought, let's make this its own rule(`use_pep604_isinstance`). It is part of PEP 604, but it's not related to _annotations_.

**Comment by martinlehoux** [comment]
Does it look better now? Maybe an idea for the violation message?

**Comment by charliermarsh** [comment]
Looks good :)

**Comment by andersk** [comment]
This only works in Python 3.10+ and needs to be conditional on `target_version >= PythonVersion::Py310`.

**Comment by charliermarsh** [comment]
Ah sorry, just an oversight on my part.

**Comment by charliermarsh** [comment]
(Hasn't gone out yet, will fix before next release.)

**Pull request merged**